vhdl_lang/src/analysis/named_entity/overloaded.rs

```diff
@@ -52,13 +66,19 @@ pub struct Signature<'a> {
     /// Vector of InterfaceObject or InterfaceFile
     pub(crate) formals: FormalRegion<'a>,
     pub(crate) return_type: Option<TypeEnt<'a>>,
+    pub(crate) generic_map: Option<Region<'a>>,
 }
 
 impl<'a> Signature<'a> {
-    pub fn new(formals: FormalRegion<'a>, return_type: Option<TypeEnt<'a>>) -> Signature<'a> {
+    pub fn new(
```

Comment:
Adding generics to the signature is a very invasive change. Are you sure this is the right approach?

Reply:
I wasn't sure what the best place to store the generics is. Putting it here felt like the right place though as the generics can be thought of as a part of the signature. I initially also tried putting it into the `Overloaded` enum (creating an `UninstSubprogram` and `UninstSubprogramDecl` variant) but that was really tedious and cumbersome compared to the current solution.

Reply:
Isnt an uninstantiated subprogram supposed to be handled very differently from other subprogams? Adding enum variants to the Overloaded enum then seems like a good way to ensure as all existing code that handles normal subprograms should not also run for uninstantiated subprograms without careful consideration.

Reply:
I have changed the implementation slightly. Now, only a `SignatureKey` has a boolean value `uninstantiated` to differentiate between uninstantiated entities. I cannot remove it there at the moment as this signature key needs to be unambiguous for overloaded entities. Do you think this approach is better?